File: ecommerce-backend/src/models/category_model.js
const { Schema, model } = require("mongoose");

const categorySchema = new Schema({
	title: { type: String, required: [true, "title is required"] },
	description: { type: String, default: "" },
	updatedAt: { type: Date },
	createdAt: { type: Date },
});

categorySchema.pre("save", function (next) {
	this.updatedAt = new Date();
	this.createdAt = new Date();

	next();
});

categorySchema.pre(
	["update", "findOneAndUpdate", "updateOne"],
	function (next) {
		const update = this.getUpdate();
		delete update._id;

		this.updatedAt = new Date();

		next();
	},
);

const CategoryModel = model("Category", categorySchema);

module.exports = CategoryModel;

File: ecommerce_app/lib/data/models/category_model.dart
class CategoryModel {
  String? sId;
  String? title;
  String? description;
  String? updatedAt;
  String? createdAt;

  CategoryModel(
      {this.sId, this.title, this.description, this.updatedAt, this.createdAt});

  CategoryModel.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    title = json['title'];
    description = json['description'];
    updatedAt = json['updatedAt'];
    createdAt = json['createdAt'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['_id'] = this.sId;
    data['title'] = this.title;
    data['description'] = this.description;
    data['updatedAt'] = this.updatedAt;
    data['createdAt'] = this.createdAt;
    return data;
  }
}
